src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
We should use descriptive property names:
```suggestion
    ? people.filter(person => person.sex === 'female' && people.find(person => person.mother === person.name))
    : people.filter((person) => person.sex === 'female');
```

You can use the same property name for every higher order function, like `.filter`, `.find` because `person` is encapsulated inside HOF and do not collide with `person` usage in other HOF